### PR TITLE
#2559. Add more constructor augmentation tests around formal parameters

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

**Review by eernstg — COMMENTED** [review]
Looks good! I made one suggestion, about a possible clarification of a description.

**Comment by sgrekhov** [comment]
Updated. 
- Added test for implicit `dynamic`.
- In `augmenting_constructors_A01_t12.dart` description  added that the type is not cpecified explicitly
- Default values removed from augmentations
PTAL.

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A01_t12.dart [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+/// @assertion It is a compile-time error if:
+/// - The signature of the constructor augmentation does not match the original
+///   constructor. It must have the same number of positional parameters, the
+///   same named parameters, and matching parameters must have the same type,
+///   optionality, and any required modifiers must match. Any initializing
+///   formals and super parameters must also be the same in both constructors.
+///
+/// @description Checks that it is not an error if a constructor augmentation
+/// specifies a type of formal parameter which were not specified in the
+/// introductory constructor.
```
[the remark this answers is not in the document]
Comment (eernstg) [reply LanguageFeatures/Augmentation-libraries/augmenting_constructors_A01_t12.dart]: Thanks!

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**